src/arrayMethodSort.js

```diff
@@ -4,8 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      for (let i = 0; i < Math.max(aStr.length, bStr.length); i++) {
+        const aChar = aStr.charCodeAt(i) || 0;
+        const bChar = bStr.charCodeAt(i) || 0;
+
+        if (aChar !== bChar) {
+          return aChar - bChar;
+        }
+      }
+
+      return 0;
+    };
+
+    const compare = compareFunction || defaultCompare;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        const comparison = compare(this[j], this[j + 1]);
+
+        if (comparison > 0) {
```

Comment:
The comparison condition is correct. You are checking if the first element is greater than the second one to decide if a swap is needed.